Spigot-Server-Patches/0230-LivingEntity-setKiller.patch

```diff
@@ -0,0 +1,30 @@
+From f6ced6606afb1efd02705d46f52280d4f4732c07 Mon Sep 17 00:00:00 2001
+From: BillyGalbreath <Blake.Galbreath@GMail.com>
+Date: Mon, 31 Jul 2017 01:49:48 -0500
+Subject: [PATCH] LivingEntity#setKiller
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
+index 3c3e8b87..640ade4f 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftLivingEntity.java
+@@ -293,6 +293,16 @@ public class CraftLivingEntity extends CraftEntity implements LivingEntity {
+         return getHandle().killer == null ? null : (Player) getHandle().killer.getBukkitEntity();
+     }
+ 
++    // Paper start
++    @Override
++    public void setKiller(Player killer) {
++        EntityPlayer entityPlayer = killer == null ? null : ((CraftPlayer) killer).getHandle();
++        getHandle().killer = entityPlayer;
++        getHandle().lastDamager = entityPlayer;
++        getHandle().lastDamageByPlayerTime = entityPlayer == null ? 0 : 100;
```

Comment:
It's been like a month, so I can't remember.. I'll be able to check this weekend.

The numbers are ticks that the killer is remembered (100 being 5 seconds which is Mojang's thing, and 0 is removing this timer if unsetting a killer). The code comes from some other NMS method that actually damages the entity, I just can't remember where off the top of my head.